internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -87,46 +87,45 @@ spec:
           TMPFILE=$(mktemp)
           PYXIS_URL="https://pyxis.engineering.redhat.com/v1"
 
-          IFS="/" read REGISTRY REPO IMAGE <<< "${1}"
-          IFS=":" read IMAGE TAG <<< "${IMAGE}"
+          IFS="/" read -r REGISTRY REPO IMAGE <<< "${1}"
+          IFS=":" read -r IMAGE TAG <<< "${IMAGE}"
 
           FETCH_URL="${PYXIS_URL}/repositories/registry/${REGISTRY}/repository/${REPO}/${IMAGE}/tag/${TAG}"
 
           # strips the last "/tag" in case $TAG is not set
           [ -z "${TAG}" ] && FETCH_URL=${FETCH_URL%/tag*}
 
-          curl --negotiate -u: "${FETCH_URL}" -o $TMPFILE
+          curl --negotiate -u: "${FETCH_URL}" -o "${TMPFILE}"
           # prints "false" in case .fbc_opt_in entry is missing
-          jq -e -r '.fbc_opt_in //false' $TMPFILE && rm -f $TMPFILE
+          jq -e -r '.fbc_opt_in //false' "${TMPFILE}" && rm -f "${TMPFILE}"
+        }
+
+        # checks if there is any previous build for the same snapshot.
+        # in case multiple builds are found, returns only the last one.
+        check_previous_prod_build() {
+          user="${1}"
+          from_index="${2}"
+          fbc_fragment="${3}"
+
+          # fetch only builds in progress or completed
```

Comment:
This means even failed ones, right? We would still pass them to the next step and they will the report that it failed, correct? Or if failed ones are not included, can you explain what makes them excluded? And yeah, we probably do want a new build if the previous failed, by default.

Related question: If there are two previous builds: First failed, second successful, will this correctly select the last (successful) one? I can't see how.

Reply:
> This means even failed ones, right? We would still pass them to the next step and they will the report that it failed, correct? Or if failed ones are not included, can you explain what makes them excluded? And yeah, we probably do want a new build if the previous failed, by default.

What came to my mind to re-add the "failed" ones was the following situation:
- task calls the IIB build;
- the *watch* task times out or fails for anything but IIB, so the build process is still running, but the task already finished
- after some minutes IIB build fails
- a new FBC pipeline starts, and  detects that the last build failed the user can look the failure.

The problem with this logic is that the next one would say that an existent build for the same fragment, so the user would need a new snapshot or a flag to rebuild the same snapshot.

> Related question: If there are two previous builds: First failed, second successful, will this correctly select the last (successful) one? I can't see how.

IIB seems to sort the builds, so the latest is 0. One option would sort the result before select the index. I will ask the IIB fellows if we can trust the output as "sorted".

Reply:
Yeah, see the other thread - they are sorted from newest to latest it seems. And you should be able to filter out the failed ones from the result.

I understand the use case for showing the failed one, but I don't think we need to worry about that - I think it should be pretty rare that it would fail, but take so long for the failure that we would time out. And even if that happens, that might mean that IIB is overloaded and so you can't really move forward with your release until it can finish builds in a reasonable time frame.

Reply:
changed back to consider only *in_progress* and *completed* builds. Failed ones will build again by default.

Reply:
sounds good